app/services/reports/list.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class Reports::List
+  include Service::Base
+
+  model :reports, optional: true
+
+  private
+
+  def fetch_reports
+    page_view_req_report_methods =
+      ["page_view_total_reqs"] +
+        ApplicationRequest
+          .req_types
+          .keys
+          .select { |r| r =~ /\Apage_view_/ && r !~ /mobile/ }
+          .map { |r| r + "_reqs" }
+
+    if !SiteSetting.use_legacy_pageviews
+      page_view_req_report_methods << "page_view_legacy_total_reqs"
+    end
+
+    reports_methods =
+      page_view_req_report_methods +
+        Report.singleton_methods.grep(/\Areport_(?!about|storage_stats)/)
+
+    reports_methods
+      .reduce([]) do |reports_acc, name|
```

Comment:
`#reduce` isn’t really needed here, we could use `#filter_map` instead. That way `nil` values will be automatically discarded and it’s just a matter of having the block returning a hash (`report_data`).